packages/hardhat-chai-matchers/test/reverted/revertedWith.ts

```diff
@@ -0,0 +1,213 @@
+        });
+      });
+    });
+
+    describe("calling a method that reverts with a panic code", function () {
+      it("successful asserts", async function () {
+        await runSuccessfulAsserts({
+          matchers,
+          method: "panicAssert",
+          successfulAssert: (x) =>
+            expect(x).to.not.be.revertedWith("some reason"),
+        });
+      });
+
+      it("failed asserts", async function () {
+        await runFailedAsserts({
+          matchers,
+          method: "panicAssert",
+          failedAssert: (x) => expect(x).to.be.revertedWith("some reason"),
+          failedAssertReason:
+            "Expected transaction to be reverted with reason 'some reason', but it reverted with panic code 0x01 (Assertion error)",
+        });
+      });
+    });
+
+    describe("calling a method that reverts with a custom error", function () {
+      it("successful asserts", async function () {
+        await runSuccessfulAsserts({
+          matchers,
+          method: "revertWithSomeCustomError",
+          successfulAssert: (x) =>
+            expect(x).to.not.be.revertedWith("some reason"),
+        });
+      });
+
+      it("failed asserts", async function () {
+        await runFailedAsserts({
+          matchers,
+          method: "revertWithSomeCustomError",
+          failedAssert: (x) => expect(x).to.be.revertedWith("some reason"),
+          failedAssertReason:
+            "Expected transaction to be reverted with reason 'some reason', but it reverted with a custom error",
+        });
+      });
+    });
+
+    describe("invalid values", function () {
+      it("non-errors as subject", async function () {
+        await expect(
+          expect(Promise.reject({})).to.be.revertedWith("some reason")
+        ).to.be.rejectedWith(AssertionError, "Expected an Error object");
+      });
+
+      it("non-string as expectation", async function () {
+        const { hash } = await mineSuccessfulTransaction(this.hre);
+
+        expect(() =>
+          // @ts-expect-error
+          expect(hash).to.be.revertedWith(10)
+        ).to.throw(
+          TypeError,
+          "Expected a string as the expected reason string"
+        );
+      });
+
+      it("errors that are not related to a reverted transaction", async function () {
+        // use an address that almost surely doesn't have balance
+        const randomPrivateKey =
+          "0xc5c587cc6e48e9692aee0bf07474118e6d830c11905f7ec7ff32c09c99eba5f9";
+        const signer = new this.hre.ethers.Wallet(
+          randomPrivateKey,
+          this.hre.ethers.provider
+        );
+
+        // this transaction will fail because of lack of funds, not because of a
+        // revert
+        await expect(
+          expect(
+            matchers.connect(signer).revertsWithoutReasonString({
+              gasLimit: 1_000_000,
+            })
+          ).to.not.be.reverted
+        ).to.be.eventually.rejectedWith(
```

Comment:
Perhaps add `ProviderError` as the first argument to this `rejectedWith()`?

Reply:
861b61f89818ae4b8a762f0cf047c8696928d215